larpandoracontent/LArMonitoring/TwoViewTransverseTracksValidationTool.cc

```diff
@@ -0,0 +1,164 @@
+/**
+ *  @file   larpandoracontent/LArMonitoring/TwoViewTransverseTracksValidationTool.cc
+ *
+ *  @brief  Implementation of the two view transverse tracks validation tool.
+ *
+ *  $Log: $
+ */
+#include "Helpers/MCParticleHelper.h"
+
+#include "larpandoracontent/LArMonitoring/TwoViewTransverseTracksValidationTool.h"
+
+#include <limits>
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+TwoViewTransverseTracksValidationTool::TwoViewTransverseTracksValidationTool() :
+    m_treeName("mytree"), 
+    m_outputFileName("output.root")
+{
+}
+
+TwoViewTransverseTracksValidationTool::~TwoViewTransverseTracksValidationTool()
+{
+    PANDORA_MONITORING_API(SaveTree(this->GetPandora(), m_treeName.c_str(), m_outputFileName.c_str(), "UPDATE"));
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TwoViewTransverseTracksValidationTool::Run(const Cluster *const pCluster1, const Cluster *const pCluster2,
+    const DiscreteProbabilityVector &discreteProbabilityVector1, const DiscreteProbabilityVector &discreteProbabilityVector2,
+    const TwoViewTransverseOverlapResult &overlapResult)
+{
+
+    TreeDataBox treeDataBox;
+
+    this->CollectTruthInformation(pCluster1, treeDataBox, "1");
+    this->CollectTruthInformation(pCluster2, treeDataBox, "2");
+
+    StoreAndRegisterDatum(static_cast<int>(this->IsSameMCParticle(pCluster1, pCluster2)), "isSameMCParticle", treeDataBox);
+
+    StoreAndRegisterDatum(static_cast<int>(discreteProbabilityVector1.GetSize()), "discProbVecSize1", treeDataBox);
```

Comment:
Usually like to see the explicit `this->` to signify a member function, but is called frequently, so might look OTT. (Subsequent discussion) If we move this to PandoraMonitoring will make context clear anyway.

Reply:
Yeah, I generally agree.  Given that it will cause me to go into an excess number of characters per line (and what you said above also applies) I might just leave as is in this instance.

Reply:
Characters per line now under control of .clang-format, so suggest do what's intrinsically best and leave the line-length management to that.